### ISSUE TITLE
get_matches() got 3 but has_match() only 1 

### ISSUE THREAD

**Issue by hcchengithub** [manoss96/pregex#26]
Doing exercise from [here](https://learnbyexample.github.io/py_regular_expressions/Exercise_solutions.html) 
### f) For the given list, filter all elements having a line starting with 'den' or ending with 'ly'.
```
items = ['love', '1\ndentist', 'fly\nfar', 'dent']
word = Indefinite(AnyLetter())
start_with_den = MatchAtLineStart("den" + word)
end_with_ly = MatchAtLineEnd(word + "ly")
pre = Either(start_with_den, end_with_ly)
```
The strange thing is that get_matches() finds 3, but . . .
```
for i, item in enumerate(items):
    print(i, pre.get_matches(item))

# 3 founds at item 1,2,3 
0 []
1 ['dentist']
2 ['fly']
3 ['dent']
```

. . . while has_match() failed with some of the 3 founds!!

```
for i, item in enumerate(items):
    print(i, pre.has_match(item), item)

# only one True at item 3 ! <------- Problem should be 3 True
0 False love
1 False 1
dentist
2 False fly
far
3 True dent
```

**Comment by manoss96** [comment]
That's definitely a bug, thanks for reporting it! I will make sure to fix it in the next version. For now, just try executing ``pre.compile()`` before executing ``pre.has_match`` so that you get the correct results :)

**Comment by hcchengithub** [comment]
Yes! Add one line has really fixed the problem.
```
pre.compile()  # <--- add one line 
for i, item in enumerate(items):
    print(i, pre.has_match(item), item)

# correct 
0 False love
1 True 1
dentist
2 True fly
far
3 True dent
```

**Comment by manoss96** [comment]
Yep! It's just that I've forgot to include some necessary RegEx flags when the pattern has not been compiled. Consider it fixed in the next version!

**Comment by manoss96** [comment]
Closing this issue as it was fixed with the release of v2.3.3
